src/hashtree.erl

```diff
@@ -413,14 +457,21 @@ width(#state{width=W}) ->
 mem_levels(#state{mem_levels=M}) ->
     M.
 
+-spec path(hashtree()) -> string().
+path(#state{path=P}) ->
+    P.
+
 %% Note: meta is currently a one per file thing, even if there are multiple
 %%       trees per file. This is intentional. If we want per tree metadata
 %%       this will need to be added as a separate thing.
--spec write_meta(binary(), binary(), hashtree()) -> hashtree().
+-spec write_meta(binary(), binary()|term(), hashtree()) -> hashtree().
 write_meta(Key, Value, State) when is_binary(Key) and is_binary(Value) ->
     HKey = encode_meta(Key),
     ok = eleveldb:put(State#state.ref, HKey, Value, []),
-    State.
+    State;
+write_meta(Key, Value0, State) when is_binary(Key) ->
```

Comment:
Thinking about it, the open/close metadata does need to be per-tree as we could crash during a partial shutdown and there's no way to know when the underlying leveldb will close.

Derp, realizing that's why Joe broke out the flush/update stuff so it could all be called before closing the database.  If you share the leveldb then as soon as the first close is called, you have no opportunity to flush/update_tree on the others.

That probably means reworking the way the opened/closed counters should work.

We didn't spot this because the unit test does not share trees.